stablehlo/dialect/StablehloOps.cpp

```diff
@@ -2878,6 +2878,12 @@ LogicalResult MapOp::inferReturnTypeComponents(
     MLIRContext*, Optional<Location> location, ValueShapeRange operands,
     DictionaryAttr attributes, RegionRange regions,
     SmallVectorImpl<ShapedTypeComponents>& inferredReturnShapes) {
+  // This function can be called directly from auto-gen builder(), and does not
+  // guarantee its arg `odsState.regions` to be non-empty. Need check it here to
+  // avoid `adaptor.getComputation()` crash.
+  if (regions.empty())
+    return emitOptionalError(location, "expect non-empty computation region");
```

Comment:
Also, can we do these checks in `inferMapOp` and `inferConditionalOp`, so that MHLO can benefit from them as well?

Reply:
Fixed.